### PR TITLE
Convert Reduce into a library node

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-dist: xenial
+dist: bionic
```
Comment (definelicht) [inline .travis.yml:1]: What does this have to do with converting reduce into a library node? ;-)

Comment (tbennun) [reply .travis.yml:1]: OpenMP implementation doesn't work unless GCC is 6/7 or newer
